src/VirtualClient/VirtualClient.Actions.UnitTests/ApacheBench/ApacheBenchExecutorTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace VirtualClient.Actions.ApacheBench
+{
+    using Moq;
+    using NUnit.Framework;
+    using System;
+    using System.CodeDom.Compiler;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using VirtualClient.Actions.Properties;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    [TestFixture]
+    [Category("Unit")]
+    public class ApacheBenchExecutorTests
+    {
+        private DependencyFixture fixture = null;
+        private DependencyPath mockPackage = null;
+
+        [Test]
+        public void ApacheBenchExecutorThrowsIfTheApacheHttpWorkloadPackageDoesNotExist()
+        {
+            this.SetupDefaultBehaviors(PlatformID.Win32NT, Architecture.X64);
+
+            using (var executor = new TestApacheBenchExecutor(this.fixture))
+            {
+                // The package does not exist on the system.
+                this.fixture.PackageManager.Clear();
+
+                DependencyException error = Assert.ThrowsAsync<DependencyException>(
+                    () => executor.InitializeAsync(EventContext.None, CancellationToken.None));
+
+                Assert.AreEqual(ErrorReason.WorkloadDependencyMissing, error.Reason);
+            }
+        }
+
```

Comment:
We can add few more tests:
1. If state of initilization is created, initialisation commands are not executed.
2. If state is not created, initialisation commands are executed